src/nccl_ofi_rdma.c

```diff
@@ -4419,10 +4394,9 @@ static int post_rdma_ctrl(nccl_net_ofi_rdma_req_t *req)
 		(freelist_regmr_fn_handle_t *)ctrl_fl_item->fl_reginfo.mr_handle;
 	nccl_net_ofi_rdma_mr_handle_t *mr_handle = fl_handle->mr_handle;
 
-	assert(xfer_info->rail_id < mr_handle->num_rails);
-	void *desc = fi_mr_desc(mr_handle->mr[xfer_info->rail_id]);
+	void *desc = fi_mr_desc(mr_handle->mr[control_rail_id]);
 
-	NCCL_OFI_TRACE_SEND_CTRL_START(req->dev_id, xfer_info->rail_id, req->comm, req, req->msg_seq_num);
+	NCCL_OFI_TRACE_SEND_CTRL_START(req->dev_id, ep->control_rail->rail_id, req->comm, req, req->msg_seq_num);
```

Comment:
This is the first patch in your series, but references code from later patches.  I don't think this series is bisectable.

Reply:
I did a late-breaking fixup commit and merged with the incorrect one, will rework history to be bisectable.